src/lcd.c

```diff
@@ -49,125 +50,104 @@ typedef struct __attribute__ ((packed)) {
+		if (Y < ((FB_HEIGHT / 8) - 0)) {
+			FB[Y][X] = temp & 0xff;
+		}
+		if (Y < ((FB_HEIGHT / 8) - 1)) {
+			FB[Y + 1][X] = temp >> 8;
+		}
 		X++;
 	}
 }
 
 void LCD_disp_str(uint8_t* theStr, uint8_t theLen, uint8_t startx, uint8_t y, uint8_t theFormat) {
 #ifdef MINIMALISTIC
-	for(uint8_t q=0;q<theLen;q++) {
-		charoutsmall( theStr[q], startx, y );
-		startx+=6;
+	for (uint8_t q = 0; q < theLen; q++) {
+		charoutsmall(theStr[q], startx, y);
+		startx += 6;
 	}
 #else
 	uint8_t invmask = theFormat & 0x80;
-	switch(theFormat & 0x7f) {
-		case FONT6X6:
-			for(uint8_t q=0;q<theLen;q++) {
-				charoutsmall( theStr[q] | invmask, startx, y );
-				startx+=6;
-			}
-			break;
-#if 0
-		case FONT12X16:
-			for(uint8_t q=0;q<theLen;q++) {
-				charoutbig( theStr[q] | invmask, startx, y );
-				startx+=12;
-			}
-			break;
-		case FONT4X6:
-			for(uint8_t q=0;q<theLen;q++) {
-				charout4x6( theStr[q] | invmask, startx, y );
-				startx+=4;
-			}
-			break;
-		case FONT6X8:
-			for(uint8_t q=0;q<theLen;q++) {
-				charout6x8( theStr[q] | invmask, startx, y );
-				startx+=6;
-			}
-			break;
-		case FONTBIGNUMBERS:
-			for(uint8_t q=0;q<theLen;q++) {
-				charoutbignum( theStr[q] | invmask, startx, y );
-				startx+=13;
-			}
-			break;
-#endif
+	for(uint8_t q = 0; q < theLen; q++) {
+		charoutsmall(theStr[q] | invmask, startx, y);
+		startx+=6;
 	}
 #endif
 }
 
 void LCD_MultiLineH(uint8_t startx, uint8_t endx, uint64_t ymask) {
-	for(uint8_t x=startx;x<=endx;x++) {
-		FB[0][x]|=ymask&0xff;
-		FB[1][x]|=ymask>>8;
-		FB[2][x]|=ymask>>16;
-		FB[3][x]|=ymask>>24;
-#if FB_HEIGHT==64
-		FB[4][x]|=ymask>>32;
-		FB[5][x]|=ymask>>40;
-		FB[6][x]|=ymask>>48;
-		FB[7][x]|=ymask>>56;
+	for (uint8_t x = startx; x <= endx; x++) {
+		FB[0][x] |= ymask & 0xff;
+		FB[1][x] |= ymask >> 8;
+		FB[2][x] |= ymask >> 16;
+		FB[3][x] |= ymask >> 24;
+#if FB_HEIGHT == 64
+		FB[4][x] |= ymask >> 32;
+		FB[5][x] |= ymask >> 40;
+		FB[6][x] |= ymask >> 48;
+		FB[7][x] |= ymask >> 56;
 #endif
 	}
 }
 
-// At the moment this is a very basic BMP file reader with the following limitations:
-// The bitmap must be 1-bit, uncompressed with a BITMAPINFOHEADER.
-uint8_t LCD_BMPDisplay(uint8_t* thebmp,uint8_t xoffset,uint8_t yoffset) {
+/*
+ * At the moment this is a very basic BMP file reader with the following limitations:
+ * The bitmap must be 1-bit, uncompressed with a BITMAPINFOHEADER.
+ */
+uint8_t LCD_BMPDisplay(uint8_t* thebmp, uint8_t xoffset, uint8_t yoffset) {
 	BMhdr_t* bmhdr;
-	uint8_t upsidedown=1;
-	uint8_t inverted=0;
+	uint8_t upsidedown = 1;
+	uint8_t inverted = 0;
 	uint16_t pixeloffset;
-	uint8_t numpadbytes=0;
+	uint8_t numpadbytes = 0;
 
 	// The following code grabs the header portion of the bitmap and caches it locally on the stack
 	BMhdr_t temp;
-	uint8_t* xxx=(uint8_t*)&temp;
-	for(uint16_t xx=0;xx<sizeof(BMhdr_t);xx++) {
-		xxx[xx]=*(thebmp+xx);
+	uint8_t* xxx = (uint8_t*) &temp;
+	for (uint16_t xx = 0; xx < sizeof(BMhdr_t); xx++) {
+		xxx[xx] = *(thebmp + xx);
```

Comment:
In fact it's not really needed at all in hardware like this as the MCU can directly read the flash (the *(thebmp+xx) is a replacement for the old "read uint16 from flash" code. Again legacy reasons where the MCU has to request program flash reads. I developed the code on an architecture where reads were allowed then suddenly ending up on a platform that didn't allow it and had to come up with a Q'n'D workaround [tm], this is the result.